File: gb-spring-backend/src/main/java/com/egroupbuy/model/PackageGroup.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Date;
import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class PackageGroup {
    int package_grp_id;
    String package_grp_name;
    Timestamp creation_date;
    int quantity;
    int product_id;
    String product_title;
    String status;
    String package_grp_image;
    Timestamp grpbuy_start;
    Timestamp grpbuy_end;
    int grpbuy_duration;
    String grpbuy_duration_type;
    int min_threshold;
    String wished;
    int package_count;
    Product product;
    List<Package> package_list;
}

File: gb-angular-frontend/src/app/interfaces/package-group.ts
import {Package} from './package';
import {Product} from './product';

export interface PackageGroup {
  package_grp_id?: number;
  package_grp_name: string;
  creation_date: string;
  quantity: number;
  product_id: number;
  product_title: string;
  status: string;
  package_grp_image: string;
  grpbuy_start: string;
  grpbuy_end: string;
  grpbuy_duration: number;
  grpbuy_duration_type: string;
  min_threshold: number;
  wished: string;
  package_count: number;
  product: Product;
  package_list: Package[];
}
